### PR TITLE
Chapter 1 Homework 3

### PR DESCRIPTION
**Pull request by Sqelson**
Tests appear to pass and program works as I intended and should fulfil the criteria, found grasping the syntax and where to put things quite difficult at first and a surprising step up from the previous homework but looking at examples of functions and classes online helped a lot.

### CODE REVIEW

File: Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs [hunk 1]
```diff
@@ -16,7 +14,7 @@ public class Lesson3Tests
         [InlineData(100, 10, 1)]
         public void CalculateBmi_With_ValidInput_Returns_Expected(float weightKg, float heightM, float expectedBmi)
         {
-            var bmi = Checks.CalculateBmi(weightKg, heightM);
```
Comment (jdphenix) [inline Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs]: You should leave the test class alone for now. Just make sure the method is called CalculateBmi (note the capitalization).

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        public static string PromptString(string message)
+        {
+            Console.WriteLine(message);
+            return Console.ReadLine();
+        }
+
+        public static float PromptFloat(string message)
+        {
+            Console.WriteLine(message);
+            return float.Parse(Console.ReadLine());
```
Comment (jdphenix) [inline Src/BootCamp.Chapter/Lesson3.cs]: Consider `float.Parse(string, IFormatProvider)` - you can pass in `CultureInfo.InvariantCulture` here.

File: Src/BootCamp.Chapter/Checks.cs [hunk 3]
```diff
@@ -16,26 +12,22 @@ public static class Checks
...
 
         public static float PromptFloat(string message)
         {
-            // To do: call your implementation. 
-            return 0;
+            return Lesson3.PromptFloat(message);
         }
 
-        public static float CalculateBmi(float weight, float height)
+        public static double CalculateBMI(float weight, float height)
```
Comment (jdphenix) [inline Src/BootCamp.Chapter/Checks.cs]: Keep this returning `float`

**Review by jdphenix — CHANGES_REQUESTED** [review]
I'm pleased with this, just a couple of minor tweaks. You shouldn't have to change the Tests project.

File: Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs (hunk 1 above)
Comment (Sqelson) [reply Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs]: Ok, I will leave the tests alone in the future until explicitly told to edit them.

File: Src/BootCamp.Chapter/Lesson3.cs (hunk 2 above)
Comment (Sqelson) [reply Src/BootCamp.Chapter/Lesson3.cs]: Took some time but figured out why this was required, and noticed it was on the wiki to implement it, thanks for the tip!

File: Src/BootCamp.Chapter/Checks.cs (hunk 3 above)
Comment (Sqelson) [reply Src/BootCamp.Chapter/Checks.cs]: Changed it back to float.

**Comment by Sqelson** [comment]
Thanks for the feedback, I have implemented the changes recommended, and also added a few comments as I realised I hadn't put any in yet.

**Review by jdphenix — APPROVED** [review]
Excellent work, thanks!

**Pull request closed without merging**